addons/website/models/website.py

```diff
@@ -292,7 +292,7 @@ def _website_api_rpc(self, route, params):
     def configurator_init(self):
         r = dict()
         company_id = self.get_current_website().company_id
-        configurator_features = self.env['website.configurator.feature'].search([])
+        configurator_features = request.env['website.configurator.feature'].search([])
```

Comment:
@detrouxdev Maybe you already decided this with @rdeodoo but should not you also fix the other `self.env` uses in the configurator code? I see one in the same function here with the call to `_website_api_rpc`, and that method also uses `self.env`

Reply:
I guess I see no reason not to change `self.env.user.lang` to `request.env.user.lang` but I was cautious to change it in other places as I'm not sure I understand when to use self.env instead request.env. Is there any moments where request would not be available?

Reply:
If you are inside a controller: use request
If you are inside a model method which is called through RPC only (like this, or simply a code encapsulation inside controllers/main.py which is only called by multiple controllers eg `_get_products_recently_viewed()` ): use request
Otherwise: use self, request might be unbound (it comes from http.py, see `request = _request_stack()`)